src/gui/profilepagewidget.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include "ocsprofileconnector.h"
```

Comment:
@FlexW want forward-declaration be enough?

Reply:
Then I would need a destructor to `ProfilePageWidget` because of `std::unique_ptr`. I really don't like forward declarations... It's time that [C++ modules](https://en.cppreference.com/w/cpp/language/modules) arrive :)

Reply:
@FlexW `ProfilePageWidget::~ProfilePageWidget = defalt;` won't work? 

> It's time that C++ modules arrive :)

Good point!

Reply:
It would, still I would have that unnecessary declaration, but I understand you point. Let's wait for @mgallien opinion :)

Reply:
> Then I would need a destructor to `ProfilePageWidget` because of `std::unique_ptr`. I really don't like forward declarations... It's time that [C++ modules](https://en.cppreference.com/w/cpp/language/modules) arrive :)

why do you not like to reduce coupling between code units using forward declarations ?

Reply:
> @FlexW `ProfilePageWidget::~ProfilePageWidget = defalt;` won't work?
> 
is the solution

Reply:
> > Then I would need a destructor to `ProfilePageWidget` because of `std::unique_ptr`. I really don't like forward declarations... It's time that [C++ modules](https://en.cppreference.com/w/cpp/language/modules) arrive :)
> 
> why do you not like to reduce coupling between code units using forward declarations ?

1. Most of the time you will use the same code units and just get annoying errors because the type is incomplete. Then you need to search for the correct C++ file. In our codebase that is easy, but it can get complex and I had issues with that in the past.
2. Just my opinion: Forward declarations are not really a feature but the necessary evil in C/C++ code. I don't know any other language that has this feature and I never missed it. I can't come up with some case where it makes really sense to switch the code units but use the same declarations. In such a case I would rather use polymorphism with virtual functions than abuse the build system for the polymorphism. If the point is to have static polymorphism then templates can be used.
3. In my opinion the only valid reason to use forward declarations is to increase compilation speed and C++ modules will help with that anyway without forward declarations.

Maybe you can give me some reasons that I overlooked and change my mind :) I will add the destructor :)